crates/bevy_reflect/src/tuple.rs

```diff
@@ -120,6 +125,60 @@ impl GetTupleField for dyn Tuple {
     }
 }
 
+/// A container for compile-time tuple info
+#[derive(Clone, Debug)]
+pub struct TupleInfo {
+    type_name: Cow<'static, str>,
+    fields: Vec<UnnamedField>,
```

Comment:
This is immutable upon construction. Capacity and the ability to extend/add fields after construction is unnecessary. Please switch to using Box<[UnnamedField]> instead. Alternatively, if it's possible to make inputs during construction `const`, a `&'static [UnnamedField]`.